### PR TITLE
add function to find submitters who should be limited

### PR DESCRIPTION
**Pull request by dvoet**
prep work for https://broadinstitute.atlassian.net/browse/GAWB-670

the idea being that the query to get the next batch of workflows to send to cromwell should exclude workflows where the submitter it retuned by this query
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Make sure liquibase is updated if appropriate
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it **(apply requires_doge label)**
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [ ] **Submitter**: Squash commits, rebase if necessary
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Merge to develop 
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala [hunk 1]
```diff
@@ -282,6 +282,25 @@ trait WorkflowComponent {
...
+     */
+    def listSubmittersWithMoreWorkflowsThan(count: Int, statuses: Seq[WorkflowStatuses.WorkflowStatus]): ReadAction[Seq[(String, Int)]] = {
+      val query = for {
+        workflow <- this if workflow.status inSetBind(statuses.map(_.toString))
+        submission <- submissionQuery if workflow.submissionId === submission.id
+      } yield (submission.submitterId, workflow)
+
+      query.
+        groupBy { case (submitter, workflow) => submitter }.
+        map { case (submitter, workflow) => submitter -> workflow.length }.
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala]: plural name: workflows or workflowRecords, etc

File: src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponentSpec.scala [hunk 2]
```diff
@@ -158,4 +158,15 @@ class SubmissionComponentSpec extends TestDriverComponentWithFlatSpecAndMatchers
     }
   }
 
+  it should "count workflows by submitter" in withDefaultTestDatabase {
```
Comment (jmthibault79) [inline src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponentSpec.scala]: Additional cases:
- intermediate number like 5
- add 2 Workflows of different terminal statuses, and `listSubmittersWithMoreWorkflowsThan(0, WorkflowStatuses.terminalStatuses)`

**Comment by jmthibault79** [comment]
Back to you

**Comment by dvoet** [comment]
comments addressed

**Comment by jmthibault79** [comment]
Go for it.

**Comment by dvoet** [comment]
I have rolled this into the main bbq PR

**Pull request closed without merging**